MBS-9506 Fix incremental room build check for Java 11

eugene-krivobokov opened the pull request.

Commit:
MBS-9506 Fix incremental room build check for Java 11

eugene-krivobokov commented:
Will fix it later. Need to override JavaVersion.current() inside Gradle test kit.
System properties `java.version` and `java.runtime.version` didn't work for this purpose.

Feel free to add a fix.

dsvoronin committed:
Merge branch 'develop' into MBS-9506-inc-room

dsvoronin committed:
Merge branch 'develop' into MBS-9506-inc-room

dsvoronin committed:
MBS-9506 try java.version

dsvoronin committed:
MBS-9506 try both

dsvoronin commented:
> Will fix it later. Need to override JavaVersion.current() inside Gradle test kit.
> System properties `java.version` and `java.runtime.version` didn't work for this purpose.
> 
> Feel free to add a fix.

Seems like you can't pass it that way
https://stackoverflow.com/a/21406600/981330

dsvoronin committed:
MBS-9506 disable failing tests

RoomProcessor uses SimpleJavaVersion that uses java.runtime.version system property to validate for incrementality

We don't know a way to mutate system properties per test in gradleTestKit

https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-master-dev/room/compiler/src/main/kotlin/androidx/room/RoomProcessor.kt
